connector_algolia/components/mapper.py

```diff
@@ -0,0 +1,20 @@
+# Copyright 2021 Akretion (http://www.akretion.com)
+# Sébastien BEAU <sebastien.beau@akretion.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+
+from odoo.addons.component.core import Component
+
+
+class AlgoliaJsonExportMapper(Component):
+    _name = "algolia.json.export.mapper"
+    _inherit = ["json.export.mapper", "algolia.se.connector"]
+
+    def _apply(self, map_record, options=None):
+        res = super()._apply(map_record, options=None)
+        # Algolia do not use the key "id" as key for the record but use "objectID"
+        # By default we set this key to the value of the id
+        # in case that you need something different you can customize your exporter
+        if "objectID" not in res:
```

Comment:
IMO you should use `_record_id_key`

Reply:
yes you right and if I use _record_id_key then the code is generic and should be moved in search engine